### DISCUSSION TITLE
Controller Change Events

### DISCUSSION THREAD

**Discussion by twiggsApproved** [spessasus/SpessaSynth#68] [Q&A] [answered]
Hi, I need to get damper/sustain on/off messages.  I've tried setting up a handler for controllerchange events, but no events are logged.  The handlers for noteon and noteoff events work as expected.  I've confirmed the midi file I'm using has damper on/off messages.  Is there something else required to receive controllerchange events and damper messages?

```js
synthesizer.eventHandler.addEvent(
	"controllerchange",
	"controllerchange",
	(e) => {
		console.log(e);
	}
);

**Comment by spessasus** [comment]
Hi,
This was a problem with the lib. It didn't send events for special ccs such as sustain, bank, or RPN controllers. I've fixed it in the latest release (3.21.13). Update to this version and the problem should go away. Remember to update `worklet_processor.min.js` too.

I hope that this helps and I apologize for the incorrect behavior with the event system.

**Comment by twiggsApproved** [comment, marked as answer]
It's working now.  Thanks!
